engine/src/main/java/io/camunda/zeebe/engine/api/TaskResult.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.api;
+
+import io.camunda.zeebe.logstreams.log.LogStreamBatchWriter;
+
+/**
+ * Here the interface is just a suggestion. Can be whatever PDT teams thinks is best to work with
+ */
+public interface TaskResult {
+
+  long writeRecordsToStream(LogStreamBatchWriter logStreamBatchWriter);
+}
```

Comment:
Yes I think this wouldn't be the end solution, since I would like to hide the writer completely but it is a great and pragmatic step forward :) :+1: